src/datalab_app_plugin_insitu/utils.py

```diff
@@ -203,29 +203,47 @@ def process_echem_data(base_folder: str, echem_folder_name: str) -> Optional[pd.
         echem_folder_path = Path(base_folder) / echem_folder_name / 'echem'
 
         if not echem_folder_path.exists():
-            warnings.warn(f"Echem folder not found at {echem_folder_path}")
-            return None
+            echem_folder_path = Path(base_folder) / echem_folder_name
+
+            if not echem_folder_path.exists():
+                warnings.warn(f"Echem folder not found at {echem_folder_path}")
+                return None
 
-        gcpl_files = [
+        mpr_files = [
             f for f in echem_folder_path.iterdir()
-            if f.suffix.upper() == '.MPR' and 'GCPL' in f.name.upper()
+            if f.suffix.upper() == '.MPR'
         ]
 
-        if not gcpl_files:
-            warnings.warn(f"No GCPL files found in {echem_folder_path}")
+        if not mpr_files:
+            warnings.warn(f"No MPR files found in {echem_folder_path}")
             return None
 
+        if len(mpr_files) == 1:
+            file_to_process = mpr_files[0]
+        else:
+
+            gcpl_files = [f for f in mpr_files if 'GCPL' in f.name.upper()]
```

Comment:
Fine to prioritise this for Jana's data, probably also need `CycleBlock.load(f)` then check if it looks a bit like GCPL or just generally timestamped data